src/types/zoneddatetime.jl

```diff
@@ -6,22 +6,20 @@ using Dates: AbstractDateTime, argerror, validargs
 # A `DateTime` that includes `TimeZone` information.
 # """
 
-struct ZonedDateTime <: AbstractDateTime
+struct ZonedDateTime{T<:TimeZone} <: AbstractDateTime
```

Comment:
I learned from Intervals.jl changing the type parameters like this should be considered a breaking change. I'd probably punt this as part of this PR

Reply:
It doesn't achieve the goal of making a ZonedDateTime with a FixedTimeZone isbits without this.

Reply:
But i could remove it from this PR.
though we would lose many of the benifits.
(we would get most of those benifits from just #327)

though it would make it easier to make a PR that pushed the timezone as a value in the type-parameter.
Which might be the preferred breaking change.